src/go/k8s/pkg/resources/license.go

```diff
@@ -0,0 +1,114 @@
+package resources
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/redpanda-data/redpanda/src/go/k8s/pkg/admin"
+)
+
+type license struct {
+	client.Client
+	scheme         *runtime.Scheme
+	cluster        *redpandav1alpha1.Cluster
+	adminAPIClient admin.AdminAPIClient
+	hash           hash.Hash
+	logger         logr.Logger
+}
+
+func NewLicense(cl client.Client, scheme *runtime.Scheme, cluster *redpandav1alpha1.Cluster, aa admin.AdminAPIClient, logger logr.Logger) *license {
+	return &license{
+		Client:         cl,
+		scheme:         scheme,
+		cluster:        cluster,
+		adminAPIClient: aa,
+		hash:           sha256.New(),
+		logger:         logger,
+	}
+}
+
+// Ensure will manages license and makes sure it is loaded in the Cluster
+func (l *license) Ensure(ctx context.Context) error {
+	if l.cluster.Spec.LicenseRef == nil || l.adminAPIClient == nil {
+		l.logger.Info(
+			"Skip ensuring license, licenseRef or adminAPI internal listener not found",
+			"licenseRef", l.cluster.Spec.LicenseRef,
+			"adminAPIClient", l.adminAPIClient,
+		)
+		return nil
+	}
```

Comment:
Could this be moved to higher log level?
Can you remove `adminAPIClient` from the log as it isn't relevant to print?
If that's not a big problem could you send real log line that this will print?
```suggestion
	if l.cluster.Spec.LicenseRef == nil || l.adminAPIClient == nil {
		l.logger.Info(
			"Skip ensuring license, licenseRef or adminAPI internal listener not found",
			"licenseRef", l.cluster.Spec.LicenseRef,
		)
		return nil
	}
```

Reply:
This is moved to debug log. I think the adminAPIClient is relevant as skip too if it is nil. The log looks like:
```
2022-11-16T06:11:40.809Z        INFO    controllers.redpanda.Cluster    Skip ensuring license, licenseRef or adminAPI internal listener not found       {"redpandacluster": "redpanda/rp-cdotvq9u2f2dgr5mfmi0", "licenseRef": {"name":"redpanda-cdotvq9u2f2dgr5mfmi0-license","namespace":"redpanda","key":"license"}, "adminAPIClient": {}}

```

Reply:
If it is not nil what it will print?